button/stories/button.stories.tsx

```diff
@@ -11,40 +11,50 @@ export default meta;
 
 export const Playground: StoryObj<ButtonProps> = {
   argTypes: {
-    children: { type: { name: 'string', required: true }, control: 'text' },
+    children: { type: { name: 'string', required: true }, control: 'text', defaultValue: 'Button' },
     variant: {
-      type: { name: 'string', required: true },
-      control: 'select',
+      type: 'string',
+      control: 'radio',
       options: ['primary', 'secondary'],
     },
-    compact: { type: { name: 'boolean', required: true }, control: 'boolean' },
-    disabled: { type: { name: 'boolean', required: true }, control: 'boolean' },
+    compact: { type: 'boolean', control: 'boolean' },
+    disabled: { type: 'boolean', control: 'boolean' },
+    icon: {
+      type: 'string',
+      options: iconNames,
+      control: 'select',
+    },
     iconPosition: {
-      type: { name: 'string', required: true },
-      options: ['none', 'left', 'right'],
-      control: { type: 'select' },
+      if: { arg: 'icon' },
+      type: 'string',
+      control: 'radio',
+      options: ['left', 'right'],
     },
-    icon: {
-      type: { name: 'string', required: false },
-      options: [iconNames[0]],
-      control: {
-        type: 'select',
-        labels: {
-          undefined: 'none',
-          left: 'left',
-          right: 'right',
+    loading: { type: 'boolean', control: 'boolean' },
+  },
+  parameters: {
+    s2d: {
+      variantProperties: [
+        'variant',
+        'compact',
+        'disabled',
+        {
+          fromArg: 'icon',
+          values: [iconNames[0]],
         },
-      },
+        {
+          fromArg: 'iconPosition',
+          values: [
+            { name: 'no icon', argValue: false }, // TODO: `undefined` (also `null`) do not work, `false` is a workaround
+            { name: 'left icon', argValue: 'left' },
+            { name: 'right icon', argValue: 'right' },
```

Comment:
@gluck isn't it a bug that these are sorted alphabetically when importing to Figma? I expect that the order will be as defined here, but I always get it sorted

![image](https://user-images.githubusercontent.com/137844/204650774-074ac212-c93c-43bd-a666-81e195507650.png)

Reply:
Not sure where this sorting is done, didn't do anything in particular for it but I agree the order should be retained